gc/base/standard/Scavenger.cpp

```diff
@@ -1647,6 +1651,11 @@ MM_Scavenger::copy(MM_EnvironmentStandard *env, MM_ForwardedHeader* forwardedHea
 			scavStats->_flipBytes += objectCopySizeInBytes;
 			scavStats->getFlipHistory(0)->_flipBytes[oldObjectAge + 1] += objectReserveSizeInBytes;
 		}
+
+		/* depth copy the hot fields of an object if scavenger dynamicBreadthFirstScanOrdering is enabled */	
+		if (_extensions->scavengerScanOrdering == MM_GCExtensions::OMR_GC_SCAVENGER_SCANORDERING_DYNAMIC_BREADTH_FIRST) {	
```

Comment:
I believe we can avoid this check and rely on _hotFieldCopyDepthCount being fixed to its max, in case of other ordering schemes.
If so, it would be wise to pull out the depth check outside of depthCopyHotFields in case the compiler does not obey inlining

Reply:
In order to rely on _hotFieldCopyDepthCount, I would need to wrap my calls to enable and disable depth copying during the finalizable pass and phantom references pass with the mentioned check for dynamic breadth first so that the depth does not get modified during the finalizable and phantom references pass, and then would just need to initially set the depth to max by default and then if dynamic breadth first scan ordering is enabled via command line, set the _hotFieldCopyDepthCount to zero. Would this solution be better than having this check here? @amicic

Reply:
@amicic

Reply:
Resolved via webex discussion.

Reply:
I see now the check added to enableHotFieldDepthCopy(), but I expected the check would be removed from here!?

Reply:
Yes my mistake, I ensured the check was not needed but forgot to remove it before pushing the changes.